src/javascript/crypto/e2e/vrf/extra25519.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * @license
+ * Copyright 2015 Yahoo Inc. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Helper functions for the Yahoo E2E vrf.verify() implementation
+ * This is ported from https://github.com/yahoo/coname/blob/master/vrf/vrf.go
+ */
+
+goog.provide('e2e.vrf.extra25519');
+
+goog.require('goog.asserts');
+goog.require('goog.crypt');
+goog.require('e2e.BigNum');
+goog.require('e2e.ecc.constant');
+goog.require('e2e.ecc.DomainParam');
+goog.require('e2e.ecc.PrimeCurve');
+
+goog.require('e2e.vrf.sha3Shake256');
+
+e2e.vrf.ed25519 = e2e.ecc.DomainParam.fromCurve(e2e.ecc.PrimeCurve.ED_25519);
+e2e.vrf.curve25519 = e2e.ecc.DomainParam.fromCurve(e2e.ecc.PrimeCurve.CURVE_25519);
+
+
+/**
+ * Unmarshals an elliptic curve point, and validates that it is an Ed25519 public key
+ * Refer to Definition 1 in https://www.iacr.org/archive/pkc2003/25670211/25670211.pdf
+ * 
+ * @param {!e2e.ByteArray} sBytes A byte array of length 32
+ * @return {!e2e.ecc.point.Ed25519}
+ */
+e2e.vrf.extra25519.fromBytesBaseGroup = function(sBytes) {
+
+	var P = e2e.vrf.ed25519.curve.pointFromByteArray(sBytes);
+
+	goog.asserts.assert(!P.isInfinity(),
```

Comment:
The checks are not optinal, and actually very important. My Go implementation was wrong. https://github.com/yahoo/coname/pull/26 is a thing that should happen (and soon after this PR it should also get tests).